pkg/new-ui/v1beta1/frontend/src/app/pages/experiment-creation/experiment-creation.component.ts

```diff
@@ -0,0 +1,133 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormArray, FormControl } from '@angular/forms';
+import { Router } from '@angular/router';
+
+import { MatDialog } from '@angular/material/dialog';
+
+import { ExperimentFormService } from '../../services/experiment-form.service';
+import { YamlModalComponent } from './yaml-modal/yaml-modal.component';
+import {
+  ExperimentK8s,
+  ExperimentSpec,
+  FeasibleSpaceMinMax,
+} from 'src/app/models/experiment.k8s.model';
+import { KWABackendService } from 'src/app/services/backend.service';
+import { NamespaceService, SnackType, SnackBarService } from 'kubeflow';
+import { pipe } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-experiment-creation',
+  templateUrl: './experiment-creation.component.html',
+  styleUrls: ['./experiment-creation.component.scss'],
+})
+export class ExperimentCreationComponent implements OnInit {
+  metadataForm: FormGroup;
+  trialThresholdsForm: FormGroup;
+  objectiveForm: FormGroup;
+  algorithmForm: FormGroup;
+  hyperParamsArray: FormArray;
+  nasGraphForm: FormGroup;
+  nasOperationsForm: FormArray;
+  metricsForm: FormGroup;
+  trialTemplateForm: FormGroup;
+  yamlTemplateForm: FormControl;
+
+  constructor(
+    private formSvc: ExperimentFormService,
+    private router: Router,
+    private dialog: MatDialog,
+    private backend: KWABackendService,
+    private ns: NamespaceService,
+    private snack: SnackBarService,
+  ) {}
+
+  ngOnInit() {
+    this.metadataForm = this.formSvc.createMetadataForm('kubeflow-user');
+    this.trialThresholdsForm = this.formSvc.createTrialThresholdForm();
+    this.objectiveForm = this.formSvc.createObjectiveForm();
+    this.algorithmForm = this.formSvc.createAlgorithmObjectiveForm();
+    this.hyperParamsArray = this.formSvc.createHyperParametersForm();
+    this.nasGraphForm = this.formSvc.createNasGraphForm();
+    this.nasOperationsForm = this.formSvc.createNasOperationsForm();
+    this.metricsForm = this.formSvc.createMetricsForm();
+    this.trialTemplateForm = this.formSvc.createTrialTemplateForm();
+    this.yamlTemplateForm = this.formSvc.createYamlTemplateForm();
+  }
+
+  /**
+   * Create an Experiment CR json obj from the form's values
+   */
+  getFormYaml(): any {
```

Comment:
Can we also add [APIVersion and Kind](https://github.com/kubeflow/katib/blob/master/examples/v1beta1/cmaes-example.yaml#L1-L2) to the showing YAML so user can copy this YAML and simple submit them later?

Reply:
Make sense, I'm on it

Reply:
pushed a commit that fixes this

![Screenshot from 2021-02-17 16-48-50](https://user-images.githubusercontent.com/11134742/108221106-1939dc00-7140-11eb-8c1d-bdffc3bfe8e2.png)